### ISSUE TITLE
[PROPOSAL] ERC1155 ERC20 wrapped tokens

### ISSUE THREAD

**Issue by adamhannigan** [sunflower-land/sunflower-land#277]
The purpose of this proposal is to explore the underlying protocols that will support the different items in the game.

**Description**

The game uses the ERC1155 multi-token standard which manages every inventory item in a single contract. This makes it easy to make contract interactions and add new items.

The ERC1155 standard keeps track of the following:
- Seeds
- Crops
- Tools
- Wood + Minerals
- Eggs, Chickens
- Rare items
- NFTs
- Food

The Sunflower Land Token is the only current ERC20 token in the game.

**Challenges**

The vision of Sunflower Land is to create a game that is integrated with real DEX economics, adding an extra layer of strategy for players.

ERC1155 standard has great adoption with NFT marketplaces like OpenSea and Rarible, however the DEX support is not great for ERC1155 tokens. This makes it difficult for people to setup liquidity pools and trade resources like Gold, Wood and Stone.

## Proposal

The following items can be withdrawn as normal and will have a decimal place of 1.

- Tools (In V1 these were 18 decimal places ERC20 tokens)
- NFTs
- Collectibles / Rare items
- Food
- Seeds
- Chickens

The following items need to go through a ERC20 token wrapper to be withdrawn. They will have 18 decimal places (makes it easier to fractionally trade on DEXes)

- Wood
- Minerals
- Crops 
- Eggs

**ERC1155 / ERC20 Wrapper**

When you go to Withdraw a resource above from your farm, we will send an ERC20 Wrapped version of the token. This will appear as a Storage House in the game where all of your crops and minerals collected live.

You would select which items to withdrawn from your farm

If you were to withdraw Wood from your farm's ERC1155 inventory, you would instead get a wrapped wood token. This is an ERC20 token that can be freely traded with any DEX supporting the ERC20 standard.

When you want to send wood to the game, you unwrap your ERC20 token (converting it back into the ERC1155 standard) and send it to your farm. This would all be behind a nice UI experience.

Essentially, no resource tokens will be in circulation. They will appear on platforms like OpenSea but the supply will be owned by the farm contracts.

**Implementation**

See the example which can be used to fork an implementation for wrapped Sunflower Land tokens

https://github.com/gnosis/1155-to-20/blob/master/contracts/Wrapped1155Factory.sol

**Comment by Stevew00dy** [comment]
My only concern here with sites like Open sea is bad actors creating fake tokens to manipulate unknowing people into paying for resources.

**Comment by adamhannigan** [comment]
@Stevew00dy Exactly. Ideally with Open Sea we will just have the 1 collection for the land NFTs and 1 collection for all in game resources, craftables etc.

With the wrapped tokens, we would just need to be clear with the community through docs which the real tokens are. Perhaps even include it as part of the UI - For instance, a miner could link you to the QuickSwap link

**Comment by j923sneaks** [comment]
I assume wrapped tokens will be unwrapped when depositing to our farm address... will unwrapping increase deposit transaction fee?

**Comment by adamhannigan** [comment]
@j923sneaks Yes, during unwrapping they would go into your farm's address. There wouldn't be any deposit fee in this case

**Comment by Z1kou** [comment]
I think most things have been adressed here I will just enforce the talk by saying that since i started playing i had no idea about the importance of ressources nor the existance of LP / Tokens dedicted for ressources , I would love to if there was a part for the ERC1155 tokens contracts and information posted in the docs of the game, giving light to this important aspect of the game to the new comers is vital to a good kickstart for those who are just starting out, on addition ofcourse to something like adding an Npc to every ressource( Example: Wood Npc is the farmer in the wood area , Miner Npc containing all Minerals contracts near the gold clomp with a cute text of him saying how much he loves gold and wishes to have the right tool to mine it , ect ...)

**Comment by adamhannigan** [comment]
@Z1kou Agreed, it adds an extra gameplay element that a lot of players don't know of. We will include some documentation

**Comment by adamhannigan** [comment]
Thanks for the contribution and feedback! ❤️ 

 We are moving towards using an ideas portal to help capture and vote on community ideas. Github will only be used for bug/technical reports. We are only a small team and it is becoming difficult for us to respond individually to every case so we have began using community voting software.   

 If you could please upload your idea to the following community portal it would be greatly appreciated 🤗   

 https://sunflower-land.ideas.aha.io/

 You can vote on similar ideas and help bring ideas to our attention. Thanks again for being apart of this community!
